sklearn/multioutput.py

```diff
@@ -636,22 +637,68 @@ def _check(self):
 
 class _BaseChain(BaseEstimator, metaclass=ABCMeta):
     _parameter_constraints: dict = {
-        "base_estimator": [HasMethods(["fit", "predict"])],
+        "base_estimator": [
+            HasMethods(["fit", "predict"]),
+            StrOptions({"deprecated"}),
+        ],
+        "estimator": [
+            HasMethods(["fit", "predict"]),
+            None,
+        ],
         "order": ["array-like", StrOptions({"random"}), None],
         "cv": ["cv_object", StrOptions({"prefit"})],
         "random_state": ["random_state"],
         "verbose": ["boolean"],
     }
 
+    # TODO(1.8): Remove base_estimator
     def __init__(
-        self, base_estimator, *, order=None, cv=None, random_state=None, verbose=False
+        self,
+        estimator=None,
+        *,
+        order=None,
+        cv=None,
+        random_state=None,
+        verbose=False,
+        base_estimator="deprecated",
     ):
+        self.estimator = estimator
         self.base_estimator = base_estimator
         self.order = order
         self.cv = cv
         self.random_state = random_state
         self.verbose = verbose
 
+    def _validate_estimator(self, default=None):
```

Comment:
I think this makes more sense as a `_get_estimator` kinda method where it returns the estimator to be used, while also validating the input wrt deprecation.

It can then be used in the `_check` function above, instead of relying on the presence of `self._estimator` attribute.

Reply:
i also used the `_get_estimator` in `_check`

```
def _check(self):
        return hasattr(self._get_estimator(), attr) or all(
            hasattr(est, attr) for est in self.estimators_
        )

```